core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -216,21 +229,80 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
     } yield clone
 
   def cloneAzureWorkspace(sourceWorkspace: Workspace,
-                          project: RawlsBillingProject,
                           profile: ProfileModel,
-                          request: WorkspaceRequest
-  ): Future[Workspace] = Future.failed(
-    RawlsExceptionWithErrorReport(
-      ErrorReport(StatusCodes.NotImplemented, "WOR-625")
-    )
-  )
+                          request: WorkspaceRequest,
+                          parentContext: RawlsRequestContext
+  ): Future[Workspace] = {
+
+    def createNewWorkspaceRecord(): Future[Workspace] =
+      dataSource.inTransaction { access =>
+        for {
+          _ <- failIfWorkspaceExists(request.toWorkspaceName)
+          workspaceId = UUID.randomUUID()
+          newWorkspace <- createMultiCloudWorkspaceInDatabase(
+            workspaceId.toString,
+            request.toWorkspaceName,
+            request.attributes,
+            access,
+            parentContext
+          )
+        } yield newWorkspace
+      }
+
+    for {
+      // The call to WSM is asynchronous. Before we fire it off, allocate a new workspace record
+      // to avoid naming conflicts - we'll erase it should the clone request to WSM fail.
+      newWorkspace <- createNewWorkspaceRecord()
+
+      cloneResult <- traceWithParent("workspaceManagerDAO.cloneWorkspace", parentContext) { context =>
+        Future(blocking {
+          workspaceManagerDAO.cloneWorkspace(
+            sourceWorkspaceId = sourceWorkspace.workspaceIdAsUUID,
+            workspaceId = newWorkspace.workspaceIdAsUUID,
+            displayName = request.name,
+            spendProfile = profile,
+            context
+          )
+        })
+      }.recoverWith { case t: Throwable =>
+        logger.warn(
+          "Clone workspace request to workspace manager failed for " +
+            s"[ sourceWorkspaceName=${sourceWorkspace.toWorkspaceName}" +
+            s", newWorkspaceName=${newWorkspace.toWorkspaceName}" +
+            s"]",
+          t
+        )
+        dataSource.inTransaction(_.workspaceQuery.delete(newWorkspace.toWorkspaceName)) >> Future.failed(t)
+      }
+
+      _ = clonedMultiCloudWorkspaceCounter.inc()
+      _ = logger.info(
+        "Created workspace record and clone job for azure workspace " +
+          s"[ workspaceId=${newWorkspace.workspaceId}" +
+          s", workspaceName=${newWorkspace.toWorkspaceName}" +
+          s", cloneJobReportId=${cloneResult.getJobReport.getId}" +
+          s"]"
+      )
+
+      // hand off monitoring the clone job to the resource monitor
+      _ <- WorkspaceManagerResourceMonitorRecordDao(dataSource).create(
+        WorkspaceManagerResourceMonitorRecord.forCloneWorkspace(
+          UUID.fromString(cloneResult.getJobReport.getId),
```

Comment:
Apparently this isn't a UUID! (see thread above)

Reply:
Thanks! Good catch

Reply:
So you know if this is a UUID for landing zones?

Reply:
ugh this is a pain because it's a binary number in the databsae.

Reply:
It's probably worth spending some time figuring out why a UUID isn't being returned-- the simplest thing is probably to actually make it a UUID.

With a lot of WSM APIs, you actually pass in the ID that you want to use. I don't know how/this Job ID is getting generated.

Reply:
@marctalbott the swagger page says that this ID is caller provided. Is there any way to make this be a UUID? Otherwise I'm gonna have to do some kind of horror in liquibase.

Reply:
@cahrens, can you give me an example? This is the first time I'm experiencing this stuff.